models/articles.models.js

```diff
@@ -63,4 +66,24 @@ exports.selectCommentFromArticleID = (article_id) => {
       }
       return data.rows;
     });
-};
\ No newline at end of file
+};
+
+exports.postCommentOnArticle = (article_id, username, body) => {
+  return db
+    .query(
+      `INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING *;`,
+      [body, username, article_id]
+    )
+    .then((data) => {
+      if (data.rows.length === 0) {
+        return Promise.reject({ status: 400, msg: "Bad Request" });
```

Comment:
I don't think that your promise.reject should be on the basis of rows being empty array.  I think it should be based on conditions similar to your model, and those error cases that you are testing for.